infrastructure/simulation-tool/src/operations/common.ts

```diff
@@ -0,0 +1,38 @@
+import { BigNumber, Signer } from 'ethers';
+import { Wallet as RollupWallet } from '@rsksmart/rif-rollup-js-sdk';
+import { depositToSelf } from '../simulations/setup';
+
+const ensureL1Funds = (funder: Signer) => async (totalDepositAmount: BigNumber, account: Signer) => {
+    const accountBalance = await account.getBalance();
+    if (accountBalance.lt(totalDepositAmount)) {
+        console.log(
+            `Funding account ${await account.getAddress()} with ${totalDepositAmount
+                .sub(accountBalance)
+                .toString()} RBTC from funder ${await funder.getAddress()}`
+        );
+
+        const latestBlock = await funder.provider.getBlock('latest');
+        const gasLimit = latestBlock.gasLimit;
+        const gasPrice = await funder.provider.getGasPrice();
+
+        const tx = await funder.sendTransaction({
+            to: await account.getAddress(),
+            value: totalDepositAmount.sub(accountBalance).mul(gasLimit).mul(gasPrice),
```

Comment:
It's not clear to me why we use the `gasLimit` and `gasPrice` to set the value

Reply:
This is done to cover the future transaction that this function is ensuring funds for

Reply:
Ok, if at some point it's not enough, we could just use a number big enough to cover everything 👍🏼